### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by cristiancavalli**
Add specific note stating the side-effects of leaving the `reportUncaughtExceptions` configuration property as `true`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +123,18 @@ var errors = require('@google/cloud-errors')({
 });
 ```
 
+#### Notes On Uncaught Exception Reporting
+
+It should be noted that leaving the `reportUncaughtExceptions` configuration option set to its default of `true`
+will cause the error-reporting library to attach to the `process` objects `'uncaughtException'` event which will
+result in the following side effects:
+
+  1. Uncaught exceptions in your script will **not** be printed to `stderr` unless ones application code has itself
```
Comment (matthewloring) [inline README.md]: I think you had written a small example of how this could be done in the original issue. Maybe it could be included here?

**Comment by ofrobots** [comment]
In offline discussions we talked about a doc with a few different options on dealing with this, including a note in the README (here). Is that still the plan?

File: README.md (hunk 1 above)
Comment (cristiancavalli) [inline README.md]: Good idea -- will include

**Comment by cristiancavalli** [comment]
@ofrobots yes, this was just meant as a stop-gap so that some literature on the subject is put front-and-center

**Pull request merged**